extensions/vertx-redis/runtime/src/main/java/io/quarkus/vertx/redis/runtime/RedisAPIsProducer.java

```diff
@@ -0,0 +1,92 @@
+package io.quarkus.vertx.redis.runtime;
+
+import javax.annotation.PreDestroy;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Produces;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import io.quarkus.vertx.redis.SyncRedisAPI;
+import io.vertx.core.Vertx;
+import io.vertx.redis.client.Redis;
+import io.vertx.redis.client.RedisAPI;
+import io.vertx.redis.client.RedisOptions;
+
+@ApplicationScoped
+public class RedisAPIsProducer {
+    static long timeout = 10;
+
+    @Inject
+    Vertx vertx;
+
+    Redis redisClient;
+
+    RedisAPI redisAPI;
+
+    SyncRedisAPI syncRedisAPI;
+
+    io.vertx.axle.redis.client.Redis axleRedisClient;
+
+    io.vertx.axle.redis.client.RedisAPI axleRedisAPI;
+
+    io.vertx.mutiny.redis.client.Redis mutinyRedisClient;
+
+    io.vertx.mutiny.redis.client.RedisAPI mutinyRedisAPI;
+
+    void initialize(RedisOptions redisOptions) {
```

Comment:
Let's try and not use this style anymore.

Recently we have updated, Data source, Mongo, Mailer and Flyway (Liquibase still needs to be fixed) to avoid this old ugly pattern.
So let's try and use the new better pattern with this extension please.

Reply:
I had this pattern hanging in for a while, but I honestly do not think it is ugly. I'll try to change it for consistency with the other.

Reply:
It's ugly in the sense that a CDI bean is created without having it's state fully populated. That goes against what CDI is meant for and was only done in the older versions of code because we had no better alternative.
With the advent of synthetic beans, we now have a much more consistent way of handling things.

Reply:
Thanks for clarifying it. I do not think we need here right? I mean, I needed the `.initialize()` method because in the earlier version of the extension there was no better way of injecting runtime config roots, now that we can I think we can make use of that. And remove other unnecessary bits. So good eye for spotting this one :-)

Reply:
Yes exactly my point. Config roots can just be infected into the beans.
Any extra options not provided by config root but created in recorders should create synthetic beans that are then injected into the beans where they are used